src/main/File/FileController.java

```diff
@@ -0,0 +1,478 @@
+    return user;
+  }
+
+  /*
+  Multipart body with following fields:
+  - "fileType": String giving the file type (see FileType enum)
+  - if "fileType" is a PDF type
+    - "toSign": boolean indicating whether or not the PDF needs signing
+    - if "toSign" is True
+      - "signature": the signature image to place in file
+  - if "fileType" is of FORM_PDF
+    - "annotated": boolean for setting whether the PDF is annotated (default false)
+    - if "annotated" is True
+      - "fileID": the fileID to replace
+  - "file": the file to be uploaded
+  - "targetUser": the user the file is being uploaded for
+   */
+  public Handler fileUpload =
+      ctx -> {
+        log.info("Uploading file...");
+        String username;
+        String organizationName;
+        UserType privilegeLevel;
+        Message response = null;
+        UploadedFile file = ctx.uploadedFile("file");
+        log.info("File: {}", file.getFilename());
+        JSONObject req;
+        String body = ctx.body();
+        try {
+          req = new JSONObject(body);
+        } catch (JSONException e) {
+          req = null;
+        }
+
+        User check = userCheck(body);
+        if (req != null && req.has("targetUser") && check == null) {
+          log.info("Target User could not be found in the database");
+          response = UserMessage.USER_NOT_FOUND;
+        } else {
+          boolean orgFlag;
+          if (req != null && req.has("targetUser") && check != null) {
+            log.info("Target user found, setting parameters.");
+            username = check.getUsername();
+            organizationName = check.getOrganization();
+            privilegeLevel = check.getUserType();
+            orgFlag = organizationName.equals(ctx.sessionAttribute("orgName"));
+          } else {
+            log.info("Checking session for user.");
+            username = ctx.sessionAttribute("username");
+            organizationName = ctx.sessionAttribute("orgName");
+            privilegeLevel = ctx.sessionAttribute("privilegeLevel");
+            orgFlag = true;
+          }
+
+          if (orgFlag) {
+            if (file == null) {
+              log.info("File is null, invalid file!");
+              response = FileMessage.INVALID_FILE;
+            } else {
+              FileType fileType = FileType.createFromString(ctx.formParam("fileType"));
+              log.info("Received file type of {}", fileType.toString());
+              String title = null;
+              boolean annotated = false;
+              boolean toSign = false;
+              if (ctx.formParam("annotated") != null) {
+                annotated = Boolean.parseBoolean(ctx.formParam("annotated"));
+              }
+              if (ctx.formParam("toSign") != null) {
+                toSign = Boolean.parseBoolean(ctx.formParam("toSign"));
+              }
+              String fileId = null;
+              UploadedFile signature = null;
+              if (fileType.isPDF()) {
```

Comment:
can we use a switch statement here? Just so we can easily tell if we need to add another file type